src/esp/gfx/RenderCamera.h

```diff
@@ -41,9 +41,41 @@ class RenderCamera : public MagnumCamera {
                                     float zfar,
                                     float hfov);
 
-  RenderCamera& draw(MagnumDrawableGroup& drawables);
+  /**
+   * @brief Overload function to render the drawables
+   * @return the number of drawables that are drawn
+   */
+  uint32_t draw(MagnumDrawableGroup& drawables);
+  /**
+   * @brief Enable or disable the frustum culling test
+   * @param true, enable; false, disable;
+   */
+  void setFrustumCullingEnabled(bool value) { frustumCullingEnabled = value; }
+  /**
+   * @brief Get the status, if the frustum culling is enabled
+   */
+  bool getFrustumCullingEnabled() { return frustumCullingEnabled; }
+
+  /**
+   * @brief performs the frustum culling
+   * @param drawableTransforms, a vector of pairs of Drawable3D object and its
+   * absolute transformation
+   * @return an iterator to the new end of the range, which contains visible
+   * pairs
+   *
+   * NOTE: user are not encouraged to call this function directly.
+   * The preferred way is to enable the frustum culling by calling @ref
+   * setFrustumCullingEnabled and then call @ref draw
+   */
+  std::vector<std::pair<std::reference_wrapper<Magnum::SceneGraph::Drawable3D>,
+                        Magnum::Matrix4>>::iterator
```

Comment:
I would really just return a `std::size_t` (`return std::remove_if(...) - drawableTransforms.begin()`), it's shorter, easier to read/document, doesn't get invalidated, and doesn't depend on the container type :)

Reply:
@mosra :
 This is a great suggestion. I agree with you that returning such a long, tedious iterator is not cool at all.
However,
- It is convenient since STL functions such as `std::erase` are defined on the iterators, and I will use the return values in such functions. It is very practical.
- I will not expect user to call this function directly as I stated clearly in the comment. It is not encouraged. "Cull()" is just a helper function. I would not even bother to extract and create it if it is not for the testing purpose. And for the same reason, I have to make it "public" instead of "protected". The point is, the only user perhaps is just me, the developer, not the user.

So if you are not strongly against it, I prefer to keep it in this way.

Reply:
Update:
I looked at this function again tonight. Yes, it is too long. I changed my mind.
Now it will return the number of visible drawables, as you suggested.
The changes afterwards were not that hard actually, which was great.